mysos/scheduler/scheduler.py

```diff
@@ -425,6 +469,18 @@ def statusUpdate(self, driver, status):
         log.error("Status update failed due to launcher error: %s" % e.message)
         self._stop()
 
+      # Update metrics.
+      # TODO(xujyan): This doesn't rule out duplicates, etc. We can consider updating these metrics
+      # in the launcher.
+      if status.state == mesos_pb2.TASK_FINISHED:
```

Comment:
And since these are terminal states that cannot transition into one another, these counters really have 1:1 relationship with tasks. Will leave them in in this PR but I am open to suggestions for improvement.